transport/internal/tlsmux/listener.go

```diff
@@ -62,6 +64,8 @@ type listener struct {
 	logger    *zap.Logger
 	mode      yarpctls.Mode
 
+	ctx         context.Context
```

Comment:
any chance we don't do this? based on the context doc: https://pkg.go.dev/context
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

Reply:
I thought about it - `net.Listener` APIs sadly didn't involve accepting context. This is similar to other transport libraries maintaining a context on the creation of connection/client, and mostly these aren't used across public APIs, for example [see](https://github.com/grpc/grpc-go/blob/13b378bc458535e6da1693fc67e09a4951cd3259/clientconn.go#L468).

A very good call though, I was able to find a way to avoid storing context. Thanks!